binderhub/builder.py

```diff
@@ -63,20 +64,40 @@ def _generate_build_name(self, build_slug, ref, limit=63, hash_length=6, ref_len
         ).lower()
 
     @gen.coroutine
+    def fail(self, message):
+        self.emit({
```

Comment:
Doesn't this need a yield here?

Reply:
yes!

Reply:
Fixed this and two other missing yields.